diagnostics/physics/sst_eval.py

```diff
@@ -17,23 +17,25 @@
 from plot_common import annotate_skill, autoextend_colorbar, corners, get_map_norm, open_var
 
 
-def plot_sst_eval(pp_root):
-    model = open_var(pp_root, 'ocean_monthly', 'tos')
-    model_grid = xarray.open_dataset('../data/geography/ocean_static.nc')
-    target_grid = model_grid[['geolon', 'geolat', 'geolon_c', 'geolat_c']].rename({'geolon': 'lon', 'geolat': 'lat', 'geolon_c': 'lon_b', 'geolat_c': 'lat_b'})
+def plot_sst_eval(pp_root, config):
+    model = open_var(pp_root, config['domain'], 'tos')
+    model_grid = xarray.open_dataset( config['model_grid'] )
+    target_grid = model_grid[config['rename_map'].keys()].rename(config['rename_map'])
     model_ave = model.mean('time').load()
 
-    glorys = xarray.open_dataset('/work/acr/mom6/diagnostics/glorys/glorys_sfc.nc')['thetao'] #.rename({'longitude': 'lon', 'latitude': 'lat'})
-    glorys_lonc, glorys_latc = corners(glorys.lon, glorys.lat)
+    glorys = xarray.open_dataset( config['glorys'] ).squeeze()['thetao'] #.rename({'longitude': 'lon', 'latitude': 'lat'})
+    try:
+        glorys_lonc, glorys_latc = corners(glorys.lon, glorys.lat)
+    except:
+        glorys_lonc, glorys_latc = corners(glorys.longitude, glorys.latitude)
     glorys_ave = glorys.mean('time').load()
     glorys_to_mom = xesmf.Regridder(glorys_ave, target_grid, method='bilinear', unmapped_to_nan=True)
     glorys_rg = glorys_to_mom(glorys_ave)
     delta_glorys = model_ave - glorys_rg
-
     oisst = (
-        xarray.open_mfdataset([f'/work/acr/oisstv2/sst.month.mean.{y}.nc' for y in range(1993, 2020)])
+        xarray.open_mfdataset([config['oisst']+f'sst.month.mean.{y}.nc' for y in range(1993, 2020)])
```

Comment:
@uwagura, do you think it would be a good idea to move the OISST and GLORYS data processing into separate functions? It might make it easier to maintain and keep things organized. We can try something like the following:

```
def process_oisst(config, target_grid, model_ave):
    """Process OISST data."""
    try:
        oisst = (
            xarray.open_mfdataset([config['oisst'] + f'sst.month.mean.{y}.nc' for y in range(1993, 2020)])
            .sst
            .sel(lat=slice(config['lat']['south'], config['lat']['north']), lon=slice(config['lon']['west'], config['lon']['east']))
        )
        oisst_lonc, oisst_latc = corners(oisst.lon, oisst.lat)
        oisst_lonc -= 360
        mom_to_oisst = xesmf.Regridder(
            target_grid,
            {'lat': oisst.lat, 'lon': oisst.lon, 'lat_b': oisst_latc, 'lon_b': oisst_lonc},
            method='conservative_normed',
            unmapped_to_nan=True
        )
        oisst_ave = oisst.mean('time').load()
        mom_rg = mom_to_oisst(model_ave)
        logger.info("OISST data processed successfully.")
        return oisst_ave, mom_rg, oisst_lonc, oisst_latc
    except Exception as e:
        logger.error(f"Error processing OISST data: {e}")
        raise
```